src/shapes/text.class.js

```diff
@@ -1343,19 +1343,12 @@
      * @return {Number} Line width
      */
     getLineWidth: function(lineIndex) {
-      if (this.__lineWidths[lineIndex]) {
+      if (this.__lineWidths[lineIndex] !== undefined) {
         return this.__lineWidths[lineIndex];
       }
 
-      var width, line = this._textLines[lineIndex], lineInfo;
-
-      if (line === '') {
-        width = 0;
-      }
-      else {
-        lineInfo = this.measureLine(lineIndex);
-        width = lineInfo.width;
-      }
+      var lineInfo = this.measureLine(lineIndex);
```

Comment:
does measureLine have the same check inside?

Reply:
Not as far as I have seen.

Reply:
When I stepped through line by line, I got as far as this._measureChar with an empty string.

Reply:
So why we want to skip the shortcut and go far down with the empty line?

Reply:
I m happy about the fix, i want to understand why we also remove the shortcut. Is not of immediate understanding to me.

Reply:
Originally I fixed the shortcut, it caused an error when a line consisted solely of a blank space because `_charBounds[x]` is undefined.

My understanding is that `this._measureLine` sets `_charBounds` for each line so we shouldn't skip it otherwise other parts of the code will fail :(.

We could restore the check but currently it's the equivalent of `if(false)` because `['']` will never equal `''`, 

Fixing it to act as a shortcut is probably possible, I'd need to investigate further and may have consequences when it comes to how different platforms measure blank newlines?.

Reply:
no is ok as it is.
The fact that skipping measurements was creating error elsewhere is good to me.